src/utils/context.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import usePersistStorage from 'react-native-use-persist-storage';
+
+export type TPersistContext<T> = {
+  data: T;
+  setData: React.Dispatch<React.SetStateAction<T>>;
+  restored: boolean;
+};
+
+export const createPersistContext = <T extends {}>({
+  defaultData,
+  storageKey,
+  version = 0,
+}: {
+  defaultData: T;
+  storageKey: string;
+  version?: number;
+}) => {
+  const createDefaultData = () => defaultData;
+
+  const Context = React.createContext<TPersistContext<T>>({
+    data: createDefaultData(),
+    setData: () => {
+      return;
+    },
+    restored: false,
+  });
+
+  const Provider: React.FC<{
+    persist?: boolean;
+  }> = props => {
+    const [data, setData, restored] = usePersistStorage<T>(
+      storageKey,
+      createDefaultData,
+      {
+        persist: props.persist,
+        version,
+      },
+    );
+
+    return (
+      <Context.Provider value={{ data, setData, restored }}>
```

Comment:
I have a new idea that
how about we return all format like `[data, setData, restored]`.

Cause now our format becomes more fixed, and the advantage is we don't need to change name every time when we get its reference.
Let user can set by his own. (like useState)

Reply:
Sure, it is great, let me update it